internal/attestation/crafter/materials/csaf.go

```diff
@@ -0,0 +1,152 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package materials
+
+import (
+	"fmt"
+
+	"github.com/openvex/go-vex/pkg/csaf"
+)
+
+// CSAFValidator is an interface for validating CSAF documents
+type CSAFValidator interface {
+	ValidateDocument() error
+}
+
+// Common structs missing on "github.com/openvex/go-vex/pkg/csaf" implementation
+
+type RevisionHistoryItem struct {
+	Date    string `json:"date"`
+	Number  int    `json:"string"`
+	Summary string `json:"summary"`
+}
+
+type Tracking struct {
+	csaf.Tracking
+	RevisionHistory []RevisionHistoryItem `json:"revision_history"`
+	Status          string                `json:"status"`
+	Version         string                `json:"version"`
+}
+
+type CSAFDocument struct {
+	csaf.CSAF
+	Document struct {
```

Comment:
Why redefining this struct? and the fields below? aren't they already in the embedded CSAF struct with the correct fields and annotations?

Reply:
Yes, the default csaf.CSAF document does not have some fields that as per the specification must exist. If you see the following line, we are doing some kind of inheritance + override of fields so we can add the missing ones.